### ISSUE TITLE
rtn undefined

### ISSUE THREAD

**Issue by Latz** [austinksmith/Hamsters.js#75]
Version: 5.3.2
Operating system: Windows 10 / Ubuntu 21.04
Platform: node v16.6.1

Example code:
```
const params = {
  array: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9],
};

hamsters.run(
  params,
  function () {
    params.array.forEach(function (i) {
      rtn.data.push(i * 2);
    });
  },
  function (output) {
    console.log('output :>> ', output);
    return output;
  }
);
```
Error: (I used the build version for disabling the logger):
```
ReferenceError: rtn is not defined
    at D:\coding\hamster\test.js:11:7
    at Array.forEach (<anonymous>)
    at Object.hamstersJob (D:\coding\hamster\test.js:10:18)
    at habitat.legacyScaffold [as legacyWheel] (D:\coding\hamster\hamsters.node.min.js:933:22)
    at pool.runTask (D:\coding\hamster\hamsters.node.min.js:1187:35)
    at pool.hamsterWheel (D:\coding\hamster\hamsters.node.min.js:1224:27)
    at D:\coding\hamster\hamsters.node.min.js:1336:27
    at new Promise (<anonymous>)
    at pool.scheduleTask (D:\coding\hamster\hamsters.node.min.js:1323:22)
    at hamstersjs.scheduleTask (D:\coding\hamster\hamsters.node.min.js:269:18)
```

Version 5.2.1 is does not contain this error.

**Comment by austinksmith** [comment]
I need to see how you are initializing the library

**Comment by Latz** [comment]
Nothing fancy:

```
hamsters.init({
  maxThreads: 4,
  debug: true,
});
```

**Comment by austinksmith** [comment]
@Latz please read the documentation on how to make use of threads in node.js, this is a bug in the legacy mode for node and I will make sure it's fixed in the next release, thank you for reporting, but you should be using the non legacy mode by following the wiki

**Comment by austinksmith** [comment]
I've just released v5.3.3 which should resolve your problem in legacy mode, marking this as resolved.

https://github.com/austinksmith/Hamsters.js/releases/tag/v5.3.3
